### PR TITLE
Added demo what is service mesh section

### LINKED ISSUE #1207: [NG] Add new sections
**Desired Situation**

Add the following, new component sections on Layer-ng:

- [x]  Do I need a service mesh?
- [x]  What is a service mesh?

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by OmkarPh**
Signed-off-by: OmkarPh <omkarphansopkar@gmail.com>

**Description**

This PR fixes #1207

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by OmkarPh** [comment]
I've added the section to home page only for Demo, have a look at it, will remove it later

**Review by Jashpatel1 — COMMENTED** [review]
The title has insufficient line-height due to which in mobile view two consecutive lines overlap. Fix that. Give some padding to subtitle as well for mobile view at least.
Screenshot from 2020-12-03 09-27-06

Also remove the changes in package-lock.json

File: src/sections/what-is-service-mesh/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import ServiceMeshWrapper from "./whatServiceMesh.style";
+
+const WhatServiceMesh = ({gradient}) => {
+    if(gradient === undefined)
+        gradient = true;
+    return (
+        <ServiceMeshWrapper gradient={gradient}>
+            <h1>What Is A Service Mesh?</h1>
+            <br/>
+            <span className="description">A dedicated layer for managing service-to-service communication.</span>
```
Comment (Jashpatel1) [inline src/sections/what-is-service-mesh/index.js:15]: @OmkarPh great work here. Just one thing, Can the font-weight be decreased a bit(make it lighter)?

Comment (OmkarPh) [reply src/sections/what-is-service-mesh/index.js:15]: Actually I didn't do any changes in code, the package-lock.json was unexpectedly showing changes, may be some issue with line endings, I just rebased my branch, will do actual changes soon

Comment (Jashpatel1) [reply src/sections/what-is-service-mesh/index.js:15]: Yaa, I noticed that afterwards😂

Comment (OmkarPh) [reply src/sections/what-is-service-mesh/index.js:15]: 😂✌️

Comment (OmkarPh) [reply src/sections/what-is-service-mesh/index.js:15]: Any other changes?
Some specific styles to be given to heading or image resolution ...

Comment (Jashpatel1) [reply src/sections/what-is-service-mesh/index.js:15]: The image takes a lot to load currently.

Comment (OmkarPh) [reply src/sections/what-is-service-mesh/index.js:15]: The other option we have is to use thubnail, https://deploy-preview-1213--layer5ng.netlify.app/static/0190594eefcb3e3c29c604706bdd6184/0d68b/The-Enterprise-path-to-service-mesh-Architectures-2nd-Edition-small.webp 

I don't think thumbnail would look good here

Comment (OmkarPh) [reply src/sections/what-is-service-mesh/index.js:15]: Is font weight ok?
You need gradient prop for the toggle right?

**Review by Jashpatel1 — APPROVED** [review]
Looks good to me. Thanks @OmkarPh

**Comment by OmkarPh** [comment]
Leave a comment before merging, I'll remove it from Home page ✌️

File: src/sections/what-is-service-mesh/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { MDXRenderer } from "gatsby-plugin-mdx";
+import { Link } from "gatsby";
+
+import bookImage from "../../collections/books/2020-10-24-the-enterprise-path-to-service-mesh-architectures-2nd-Edition/The-Enterprise-path-to-service-mesh-Architectures-2nd-Edition.png";
```
Comment (Tanuj22) [inline src/sections/what-is-service-mesh/index.js:5]: Keep this image in the same folder so it's easier to keep track of

**Review by Tanuj22 — COMMENTED** [review]
Just a small change and we are good to go

File: src/sections/what-is-service-mesh/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { MDXRenderer } from "gatsby-plugin-mdx";
```
Comment (Tanuj22) [inline src/sections/what-is-service-mesh/index.js:2]: Why import this?

Comment (OmkarPh) [reply src/sections/what-is-service-mesh/index.js:2]: It's useless, thought of using it for the image earlier 🤦🏻‍♂️😂

**Review by Tanuj22 — APPROVED** [review]
Nicely done @OmkarPh :)

**Pull request merged**